test/GPv2Settlement.test.ts

```diff
@@ -127,4 +130,124 @@ describe("GPv2Settlement", () => {
       ).to.be.revertedWith("invalid pair");
     });
   });
+
+  describe("verifyClearingPrice", () => {
+    let uniswapPair: Contract;
+
+    beforeEach(async () => {
+      uniswapPair = await waffle.deployMockContract(
+        deployer,
+        IUniswapV2Pair.abi,
+      );
+    });
+
+    it("should allow clearing prices within Uniswap spot price range", async () => {
+      await uniswapPair.mock.getReserves.returns(1e6, 2e8, 0);
+      for (const [clearingPrice0, clearingPrice1] of [
+        [1000, 200000],
+        [997, 199800], // NOTE: Minimum price
+        [999, 199400], // NOTE: Maximum price
+      ]) {
+        await expect(
+          settlement.verifyClearingPriceTest(
+            uniswapPair.address,
+            0,
+            0,
+            clearingPrice0,
+            clearingPrice1,
+          ),
+        ).to.not.be.reverted;
+      }
+    });
+
+    it("should revert if clearing price is outside of Unswap spot price range", async () => {
+      await uniswapPair.mock.getReserves.returns(1e6, 2e8, 0);
+      await expect(
+        settlement.verifyClearingPriceTest(
+          uniswapPair.address,
+          0,
+          0,
+          10021,
+          2000000,
+        ),
+      ).to.be.revertedWith("Uniswap price not respected");
+      await expect(
+        settlement.verifyClearingPriceTest(
+          uniswapPair.address,
+          0,
+          0,
+          9999,
+          2004000,
+        ),
+      ).to.be.revertedWith("Uniswap price not respected");
+    });
+
+    it("should allow clearing prices within Uniswap effective price range", async () => {
+      // NOTE: If `d0` is negative, then we are removing token 0 from the
+      // Uniswap reserves, i.e. we are buying token 0 for token 1. Conversely,
+      // if `d1` is negative, then we are buying token 1 for token 0.
+
+      for (const [clearingPrice0, clearingPrice1] of [
+        [1000, 199800], // NOTE: Minimum price
+        [9990, 1988018], // NOTE: Maximum price
+      ]) {
+        await expect(
+          settlement.verifyClearingPriceTest(
+            uniswapPair.address,
+            -1,
+            200,
+            clearingPrice0,
+            clearingPrice1,
+          ),
+        ).to.not.be.reverted;
+      }
+
+      for (const [clearingPrice0, clearingPrice1] of [
```

Comment:
ubernit: maybe split these two test cases in separtate `it` blocks
a) when sellToken is sold to Uniswap
b) when buyToken is sold to Uniswap